### PR TITLE
chore: change user-agent of fetch headers

### PR DESCRIPTION
**Pull request by enguerran**
this is used on cozy-stack to enhance statistics
https://trello.com/c/mg9hDsFD/935-fiabiliser-les-stats-des-apps-mobile

### CODE REVIEW

**Comment by enguerran** [comment]
[(?) Fiabiliser les stats des apps mobile](https://trello.com/c/mg9hDsFD/935-fiabiliser-les-stats-des-apps-mobile)

**Comment by y-lohse** [comment]
The code LGTM, I'll try to run it to make sure it works for me to today.

**Comment by enguerran** [comment]
LMK if it works for you.

**Comment by enguerran** [comment]
> However, have you been able to confirm it actually changes the user agent header with a mobile device?

No, I haven't

**Comment by enguerran** [comment]
It appears the first solution failed to be stable. I adopted another solution recommended by ionic developers: https://ionicframework.com/docs/native/user-agent/
It works better and should allow us to do some statistics

File: src/drive/mobile/lib/cozy-helper.js [hunk 1]
```diff
@@ -91,3 +91,27 @@ const getTokenWithNoException = async () => {
...
+const getUserAgent = navigator => navigator && navigator.userAgent
+
+export const updateUserAgent = () => {
+  try {
+    window.UserAgent.set(getUpdatedUserAgent(window.navigator))
+  } catch (err) {
+    console.error(
+      `Unable to update User-Agent with cordova plugin: ${err.message}`
+    )
+    throw err
```
Comment (y-lohse) [inline src/drive/mobile/lib/cozy-helper.js]: Do we really want to `throw` here? There's no other catch block down the line, and this code runs at the start of the application, so everything will fail if the user agent can't be set. That seems a bit extreme no?

Comment (enguerran) [reply src/drive/mobile/lib/cozy-helper.js]: We must fail fast to be as reliable as possible.

This is a library function called by someone. This someone should handle the exception if the user-agent were not updated. Where you are right is that we must handle this exception in the [`startApplication` function that makes the call](https://github.com/cozy/cozy-drive/pull/895/files/445f4bd2e3942280eb01527fc554c7283b6d43d8#diff-f112ba1b7dce001913124514eee437fbR135).

**Pull request merged**